_posts/2019-04-24-accessing-kafka-part-2.md

```diff
@@ -0,0 +1,317 @@
+
+# Customizations
+
+Strimzi aims to make node ports work out of the box.
+But there are several options which you can use to customize the Kafka cluster and its node port services.
+
+## Pre-configured node port numbers
+
+By default, the node port number are generated / assigned by the Kubernetes controllers.
+That means that every time you delete your Kafka cluster and deploy a new one, a new set of node ports will be assigned to the Kubernetes services created by Strimzi.
+So after every redeployment you have to reconfigure all your applications using the node ports with the new node port of the bootstrap service.
+
+Strimzi allows you to customize the node ports in the `Kafka` custom resource:
+
+```yaml
+# ...
+listeners:
+  external:
+    type: nodeport
+    tls: true
+    authentication:
+      type: tls
+    overrides:
+      bootstrap:
+        nodePort: 32100
+      brokers:
+      - broker: 0
+        nodePort: 32000
+      - broker: 1
+        nodePort: 32001
+      - broker: 2
+        nodePort: 32002
+# ...
+```
+
+The example above requests node port `32100` for the bootstrap service and ports `32000`, `32001` and `32002` for the per-broker services.
+This allows you to re-deploy your cluster without changing the node ports numbers in all your applications.
+
+However, Strimzi doesn't do any validation of the requested port numbers.
+So you have to make sure that they are:
+
+* within the range assigned for node ports in the configuration of your Kubernetes cluster
+* not used by any other service.
+
+You do not have to configure all the node ports.
+You can decide to configure only some of them - for example only the one for the external bootstrap service.
+
+## Configuring advertised hosts and ports
+
+Strimzi also allows you to customize the advertised hostname and port which will be used in the configuration of the Kafka pods:
+
+```yaml
+# ...
+listeners:
+  external:
+    type: nodeport
+    authentication:
+      type: tls
+    overrides:
+      brokers:
+      - broker: 0
+        advertisedHost: example.hostname.0
+        advertisedPort: 12340
+      - broker: 1
+        advertisedHost: example.hostname.1
+        advertisedPort: 12341
+      - broker: 2
+        advertisedHost: example.hostname.2
+        advertisedPort: 12342
+# ...
+```
+
+The `advertisedHost` field can contain either DNS name or an IP address.
+You can of course also decide to customize just one of these.
+
+Changing the advertised port will only change the advertised port in the Kafka broker configuration.
+It will have no impact on the node port which is assigned by Kubernetes.
+To configure the node port numbers used by the Kubernetes services, use the `nodePort` option described above.
+
+Overriding the advertised hosts is something we already used in one of the previous chapters when the host provided by the Kubernetes API was not the correct one.
```

Comment:
"Previous chapers" doesn't really make sense in a blog post. I think you mean sections, right?